commit ba0a7533def990ae0cf481992d6b37a558b4f8ff
Author: Tim Foster <tim.foster@joyent.com>
Date:   2019-08-13T13:59:23+01:00 (8 weeks ago)
    
    OS-7921 SmartOS bash_completions should complete alias names
    Reviewed by: John Levon <john.levon@joyent.com>

diff --git a/src/fw/etc/fwadm.completion b/src/fw/etc/fwadm.completion
index 5076c368..939f69cf 100644
--- a/src/fw/etc/fwadm.completion
+++ b/src/fw/etc/fwadm.completion
@@ -17,6 +17,9 @@ _fwadm()
         vms_uuids=$(zoneadm list -cp | grep -v ':global:' | cut -d':' -f5 | \
             sort | uniq)
         COMPREPLY=( $(compgen -W "${vms_uuids}" -- ${cur}) )
+        if [[ -z "$COMPREPLY" ]]; then
+            _zone_alias
+        fi
 
     elif [[ ${prev} == 'delete'
           || ${prev} == 'disable'
diff --git a/src/img/etc/imgadm.completion b/src/img/etc/imgadm.completion
index 44e3de82..228ed250 100644
--- a/src/img/etc/imgadm.completion
+++ b/src/img/etc/imgadm.completion
@@ -85,6 +85,9 @@ _imgadm()
         # VM UUIDs.
         local vm_uuids=$(vmadm list -H -o uuid)
         COMPREPLY=( $(compgen -W "${vm_uuids}" -- ${cur}) )
+        if [[ -z "$COMPREPLY" ]]; then
+            _zone_alias
+        fi
 
     else
         # Just expand files by default.
diff --git a/src/vm/etc/vmadm.completion b/src/vm/etc/vmadm.completion
index 89871e13..6d9f7f85 100644
--- a/src/vm/etc/vmadm.completion
+++ b/src/vm/etc/vmadm.completion
@@ -27,6 +27,9 @@ _vmadm()
         vms_uuids=$(zoneadm list -cp | grep -v ':global:' | cut -d':' -f5 | \
             sort | uniq)
         COMPREPLY=( $(compgen -W "${vms_uuids}" -- ${cur}) )
+        if [[ -z "$COMPREPLY" ]]; then
+            _zone_alias
+        fi
 
     elif [[ ${prev} == 'info'
           || ${prev} == 'sysrq' ]]; then
@@ -36,6 +39,9 @@ _vmadm()
             grep ":[bh]*[ky]v[em]:excl:" | \
             cut -d':' -f5 | sort | uniq)
         COMPREPLY=( $(compgen -W "${vms_uuids}" -- ${cur}) )
+        if [[ -z "$COMPREPLY" ]]; then
+            _zone_alias
+        fi
 
     elif [[ ${prev} == 'create-snapshot'
         || ${prev} == 'delete-snapshot'
@@ -44,6 +50,9 @@ _vmadm()
         vms_uuids=$(zoneadm list -cp |
 	    nawk -F: '$2 != "global" && $6 != "kvm" { print $5 }' | sort -u)
         COMPREPLY=( $(compgen -W "${vms_uuids}" -- ${cur}) )
+        if [[ -z "$COMPREPLY" ]]; then
+            _zone_alias
+        fi
 
     elif [[ ${prev} == 'validate' ]]; then
 
